rust/osm2lanes-web/src/main.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::BTreeMap;
+
+use serde::{Deserialize, Serialize};
+use web_sys::HtmlInputElement;
+use yew::prelude::*;
+
+use osm2lanes::{get_lane_specs_ltr, Config, Direction, DrivingSide, LaneSpec, LaneType};
+
+// Use `wee_alloc` as the global allocator.
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+type ShouldRender = bool;
+
+struct WebLaneType(LaneType);
+
+impl WebLaneType {
+    /// Represents the lane type as a single character. Always picks one buffer type.
+    pub fn to_char(self) -> char {
+        match self.0 {
+            LaneType::Driving => '🚗',
+            LaneType::Biking => '🚲',
+            LaneType::Bus => '🚌',
+            LaneType::Parking => '🅿',
+            LaneType::Sidewalk => '🚶',
+            LaneType::Shoulder => 'ˢ',
+            LaneType::SharedLeftTurn => '🔃',
+            LaneType::Construction => 'x',
+            LaneType::Buffer(_) => '|',
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct State {
+    pub edit_value: String,
+    pub lanes: Vec<LaneSpec>,
+}
+
+pub enum Msg {
+    Submit(String),
+    Focus,
+}
+
+pub struct App {
+    focus_ref: NodeRef,
+    state: State,
+}
+
+fn string_to_tags(input: &str) -> Result<BTreeMap<String, String>, String> {
+    let mut map = BTreeMap::new();
+    for line in input.lines() {
+        let (key, val) = line.split_once("=").ok_or("tag must be = separated")?;
+        map.insert(key.to_owned(), val.to_owned());
+    }
+    Ok(map)
+}
+
+const CFG: Config = Config {
+    driving_side: DrivingSide::Right,
```

Comment:
A possible next step (doesn't have to be in this PR) could be to add a checkbox or toggle for this

Reply:
yes, a next step please :)